jira/client.py

```diff
@@ -2828,15 +2830,26 @@ def search_users(
         :type includeActive: bool
         :param includeInactive: If true, then inactive users are included in the results. (Default: False)
         :type includeInactive: bool
+        :param query: Search term. It can just be the email.
+        :type query: str
 
 
         :rtype: ResultList
         """
+        if not user and not query:
+            raise ValueError("Either 'user' or 'query' arguments must be specified.")
+
         params = {
             "username": user,
             "includeActive": includeActive,
             "includeInactive": includeInactive,
         }
+
+        if user:
+            params["username"] = user
```

Comment:
@julenpardo Thanks for your efforts on this stuff! We are looking forward for it will be merged! :heart: 
Just one question about changes, isn't it redundant to add this check if the `username` parameter is already in `params` dictionary ?
https://github.com/pycontribs/jira/blob/b5ef5288516117cae6702c54a6d066fe271afc08/jira/client.py#L2843

Reply:
Hey @darlev91, thanks for your feedback! You're absolutely right, thanks for the catch! I'll push the change asap

Reply:
@darlev91 I've already made the changes, thanks again for the feedback!

Reply:
@julienadriano Thanks a lot!
What's about `query` parameter ? It is compatible with both JIRA API versions. I think it's better to keep things simple

Reply:
@darlev91 is it? This is actually what @ngenovictor pointed out above. So we can just use the `query` parameter right away and keep it backwards compatible?

I did it this way just to be 100% sure that it was compatible with both versions, as I can only test it with the latest one, so I would not be sure about replacing the `username` parameter. But if that's the case, then I can just apply @ngenovictor's suggestion

Reply:
@julienadriano yes, if checking JIRA API docs, it says that `query` parameter is required (can contain not only `emailAddress` as a value, and `displayName` too), but it shouldn't replace `username` because it is separate one:
[Link on JIRA API v3](https://developer.atlassian.com/cloud/jira/platform/rest/v3/?utm_source=%2Fcloud%2Fjira%2Fplatform%2Frest%2F&utm_medium=302#api-rest-api-3-user-search-get)
[Link on JIRA API v2](https://developer.atlassian.com/cloud/jira/platform/rest/v2/#api-rest-api-2-user-search-get)
I've checked it already and it really works fine for both API versions, so I think it will be better to add it directly in parameters like separate parameter that should be send on API.

Reply:
@julienadriano @ngenovictor I think this point of view make sense in this case:
```
def search_users(self, query, username=None, startAt=0, maxResults=50, includeActive=True, includeInactive=False):
     ...
     params = {
            "query": query,
            "username": username,
            "includeActive": includeActive,
            "includeInactive": includeInactive,
        }
        ...
```
In this case the client code shouldn't be broken if it uses `displayName` or `emailAddress` as a `query` 
What do you think about it, guys ?

Reply:
@darlev91 Makes sense to me, I've already pushed the changes